multicluster/cmd/link.go

```diff
@@ -204,9 +205,13 @@ A full list of configurable values can be found at https://github.com/linkerd/li
 				return err
 			}
 
-			gatewayPort, err := extractGatewayPort(gateway)
-			if err != nil {
-				return err
+			gatewayPort := opts.gatewayPort
+			if gatewayPort == 0 {
+				serviceGatewayPort, err := extractGatewayPort(gateway)
```

Comment:
Can we not assign to `gatewayPort` directly here since we are returning on err anyway?
```suggestion
				gatewayPort, err := extractGatewayPort(gateway)
```

Reply:
I have implemented the suggestion of @dadjeibaah